### ISSUE TITLE
Directory Detail Subpane: how to enable?

### ISSUE THREAD

**Issue by tbenbow** [sodal-project/cartographer#106]
When a row is clicked in either of the directory tables a modal opens with details about the persona that was clicked.

The modal that opens is a new module called "Detail Pane". This module should be allowed to be used by other modules beyond the directory module. This is not functionality that is unique to the directory.

The "Detail Pane" module has the following requirements...

- A close button in the upper right corner.
- A heading area with...
  - The UPN and a button to copy it to the clipboard.
  - The name which is one of the following: persona.Name OR persona.friendlyName OR personaId
  - A content area which can contain other modules

Inside the content area of the detail pane we will have several other modules/component...

1. A layout module that will control the placement of other modules. This may be a one off module specifically for the directory module or perhaps this layout comes from the directory module.

2. A Persona Grid module with Key/Value pair cards:
    - All persona properties except: platform, type, id and upn
    - There will be a database call to core.getPersona that gives us these properties

3. A Notes module with a large textarea containing directory notes.
    - This has a save button.
    - This is just stubbed out for now, it doesn't retrieve or save data

4. A tab row to control the visibility of three tables:  "Aliases", "Controls" and "Obeys"

5. Each tab area contains a personaTable
      - Draw all the tables on initialization
      - Each time a save happens from the table filters...
        - It will refresh just the target table
        - It will maintain the filter state for each persona
      - Andy will provde for each of these tables configs
      - Inside of the config there is a "table-form-id"
      - When you update the filters it saves their state to the personaTable module namespace in the config database.

In summary we will have the following new modules...

- Detail Pane
- Directory Detail Pane Layout (maybe this is included in the directory module)
- Persona Tables:  The "Aliases", "Controls" and "Obeys". (maybe in defined in the directory module or it could be it's own module with or without the tabs.

And the following new components...

- Tabs: This needs to control the visibility of the dom elements passed in and have a content slot that will contain all of the markup it's responsible for showing and hiding.

**Comment by andyschwab** [comment]
Some clarification and definition notes:

* "module" means a canonical Cartographer Module
* "component" means a Handlebars component
* "panes", or "subpanes", refere to UI elements delivered by modules

_TODO_

**1 - Detail Pane Module** 

The Detail Pane Module is a Cartographer Module that enables the DetailPane Handlebars component, as well as all DetailPane configuration processing.

The Detail Pane Module provides the DetailPane partial, which generates a complete DetailPane module view when passed a DetailPane build object.

`{{ DetailPane data=buildObject }}`

The Detail Pane Module has two exported functions:
- `build(config)` -> accepts a DetailPane configuration object, and returns a valid **buildObject**
- `update(detailForm)` -> accepts a detailForm request and responds with updated DetailPane markup

The minimal DetailPane **buildObject** is: 
``` { upn, } ```

When build(config) is called, the DetailPane Module:
1) Loops through core.mod.<modules>. and calls all "getDetailSubpane(upn)" 
2) Generates a buildConfig that will create and populate related subpanes, i.e.:

```

subpaneResponse = core.mod.directory.getDetailSubpane(upn)

buildObject = {
  upn: `upn:email:account:test@example.com`,
  displayName: `Example Email`
  subpanes: [{
      component: subpaneResponse.component
      data: subpaneResponse.data
    }]
}
```

**2 - Directory Subpane Call**

`core.mod.directory.getDetailSubpane(upn)`

This will return a subpane configuration object that provides the Component template name and the configuration data to build that subpane. For the Directory Subpane Call, that means:

* {{ DirectoryDetailSubpane }}, with the complete layout for the subpane
* Support for your items 1 - 5; I imagine something like:

```
{{ DirectoryDetailSubpane }}
  {{ DirectoryPersona data=persona }}
  {{ DirectoryTableTabs data=tables }}
    {{ PersonaTable data=tables.aliasTable }}
    {{ PersonaTable data=tables.controlTable }}
    {{ PersonaTable data=tables.obeyTable }}
```

**3 - Implement Detail Pane within personaTable**

Generate a Detail Pane whenever a row is clicked in the personaTable. This precludes the need for any additional work in Directory to support the Detail Pane. The functionality will carryover from Directory's use of personaTable.
